frontend/src/components/NotFound.test.tsx

```diff
@@ -0,0 +1,30 @@
+import { describe, expect, test } from "vitest";
+import { render, screen } from "@testing-library/react";
+import { MemoryRouter, Route, Routes } from "react-router";
+import NotFound from "./NotFound";
+
+interface NotFoundTestParams {
+  path: string;
+  expectedText: string;
+}
+
+const testCases: NotFoundTestParams[] = [
+  {
+    path: "/random-path",
+    expectedText: "404 Not Found",
+  },
+];
+
+describe.each(testCases)("NotFound Component", ({ path, expectedText }) => {
+  const testName = `Render correctly when navigating to '${path}'`;
+  test(testName, () => {
+    render(
+      <MemoryRouter initialEntries={[path]}>
```

Comment:
I don't think this test is good. If I understand it correctly, what you've done is set up a *new* router which always renders the `NotFound` component, and then browsed to some path (`/random-path` -- but even `/` would have the same behavior!) and asserted the 404 text is shown.

That isn't useful.

You should test *our* application renders the `NotFound` component on a route not known by *our* application, otherwise we're testing something completely unconnected to our application, which could be entirely broken (say if you forgot to make the change to `index.tsx` -- this test would still pass).